src/server/world.zig

```diff
@@ -927,6 +939,36 @@ pub const ServerWorld = struct { // MARK: ServerWorld
 		self.dropWithCooldown(stack, pos, dir, velocity, 0);
 	}
 
+	fn tickBlocksInChunk(self: *ServerWorld, _chunk: ?*chunk.ServerChunk) void {
+		if(_chunk) |ch| {
+			for(0..self.tickSpeed) |_| {
+				const blockIndex: u32 = main.random.nextInt(u32, &main.seed);
+
+				const x: i32 = @intCast(blockIndex >> chunk.chunkShift2 & chunk.chunkMask);
+				const y: i32 = @intCast(blockIndex >> chunk.chunkShift & chunk.chunkMask);
+				const z: i32 = @intCast(blockIndex & chunk.chunkMask);
+
+				var block = ch.super.getBlock(x, y, z);
+				for(block.tickEvents()) |event| {
+					event.tryRandomTick(block, ch, x, y, z);
+				}
+			}
+		}
+	}
+
+	fn tick(self: *ServerWorld) void {
+		// tick blocks
+		var iter = ChunkManager.entityChunkHashMap.keyIterator();
+		while(iter.next()) |pos| {
```

Comment:
This is not thread safe, another thread could modify the hashmap while you are iterating.